### ISSUE TITLE
Chapter 3 CrudRepository Tests Fail.

### ISSUE THREAD

**Issue by ezazpi** [spring-boot-in-practice/repo#2]
(repo/ch03/manage-domain-objects-with-jpa/course-tracker-final/)

My code matches your GitHub code except I'm updating the source that I started with in chapter 1.  As a result you'll see ch01 instead of ch03 in the path.  Everything compiles, but the tests fail.  My Test has:

```
    @Autowired
    private CourseRepository courseRepository;
```

My CourseRepository matches your GitHub code (only a @repository annotation).

```
package com.manning.sbip.ch01.repository;

import com.manning.sbip.ch01.model.Course;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface CourseRepository extends CrudRepository<Course, Long> {
}
```

I also updated Course to match the JPA changes provided in the GitHub repository.

At the start of the test output, I see (notice no JPA repository interfaces are found):

`2021-07-24 13:12:29.041 [ INFO] [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 21 ms. Found 0 JPA repository interfaces.`

I also tried updating SpringBootAppDemoApplicationTests with Autowired required=true, but it did not help.

    @Autowired(required=true)
    private CourseRepository courseRepository;

```
2021-07-24 12:39:51.334 [ERROR] [           main] o.s.t.c.TestContextManager               : Caught exception while allowing TestExecutionListener [org.springframework.boot.test.autoconfigure.SpringBootDependencyInjectionTestExecutionListener@10e4cc6] to prepare test instance [com.manning.sbip.ch01.springbootappdemo.SpringBootAppDemoApplicationTests@1e203e6]
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'com.manning.sbip.ch01.springbootappdemo.SpringBootAppDemoApplicationTests': Unsatisfied dependency expressed through field 'courseRepository'; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'com.manning.sbip.ch01.repository.CourseRepository' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true)}
```
I checked for obvious package name problems, but the source matches yours except for the ch01 references.  There are no compile issues.

Running the app using `mvn spring-boot:run` is successful, but the same `Found 0 JPA repository interfaces` message is displayed.
 
I am running on Linux and using PostgreSQL instead of H2, but I would not think either would cause this problem.  The previous relational tests worked fine.  It's just this switch to JPA that caused the issue.

**Comment by ezazpi** [comment]
I cloned the project and started with course-tracker-final.  I tried it as is (H2 database) and it worked.  I then brought in my application.properties and pom.xml to support postgresql.  I did have to include spring.datasource.initialization-mode=always and also had spring.jpa.hibernate.ddl-auto=create in my application.properties file.  Repeated tests worked fine.  Using spring.datasource.initialization-mode=always seems to be the only difference from what didn't work.  Previously, my editor told me spring.datasource.initialization-mode=always was deprecated and a search indicated I should use spring.sql.init.mode=always.  Both property names were unrecognized previously, but now spring.datasource.initialization-mode=always is recognized.  I had been iteratively building upon the original code since chapter 1.  For some reason the switch to JPA caused issues.  Maybe a clean will help as well, but other than things like Log4j2 and validation, the original code doesn't look any different than the cloned code.  Now, though, the repository interface is found.
